test/WaveOps/GroupMemoryBarrierWithGroupSync.test

```diff
@@ -0,0 +1,100 @@
+#--- source.hlsl
+RWStructuredBuffer<uint4> Out : register(u0);
+
+groupshared uint4 SharedData;
+groupshared uint4 Indices[128];
+
+// Note: Placing GroupMemoryBarrierWithGroupSync in divergent control branches
+// is undefined, and hence, untested
+
+[numthreads(128,4,1)]
+void main(uint3 ThreadID : SV_GroupThreadID) {
+
+  // Basic Broadcast
+  if (ThreadID.x == 127 && ThreadID.y == 3) {
+    SharedData = 1;
+  }
+
+  // Prevents SharedData being read below before being initialized
+  GroupMemoryBarrierWithGroupSync();
+
+  if (ThreadID.x == 0) {
+    Out[0][ThreadID.y] = SharedData[ThreadID.y];
+  }
+
+  // Prevents SharedData being updated below before written to Out[0]
+  GroupMemoryBarrierWithGroupSync();
+
+  // Interlocked Accumulation
+  for (uint I = 0; I < 128; I++) {
+    if (ThreadID.x == I) {
+      SharedData[ThreadID.y] = SharedData[ThreadID.y] + 1;
+    }
+
+    // Prevents SharedData datarace across ThreadID.x, and,
+    // SharedData being written before fully accumulated
+    GroupMemoryBarrierWithGroupSync();
+  }
+
+  if (ThreadID.x == 127) {
+    Out[1][ThreadID.y] = SharedData[ThreadID.y];
+  }
+
+  // Strided Read/Write:
+  Indices[ThreadID.x][ThreadID.y] = ThreadID.x + ThreadID.y * 128;
```

Comment:
Do we really need any more tests than we have after `Out[1]`?  Arguably `Out[0]` is enough too, but it does seem that the other one is a bit more interesting.